### ISSUE TITLE
Bastion with user SSH keys from IAM fails with user names in the form of...

### ISSUE THREAD

**Issue by apatrida** [widdix/aws-cf-templates#63]
Bastion with user SSH keys from IAM fails with user names in the form of...
`some.person@mydomain.com`

gives an error for `useradd` as:

> useradd: invalid user name some.person.at.mydomain.com

**Comment by michaelwittig** [comment]
TL;DR could it be that "some.person.at.mydomain.com" is in real longer than 32 characters?

Background:

Allowed characters for IAM user names are:
> alphanumeric, including the following common characters: plus (+), equal (=), comma (,), period (.), at (@), underscore (_), and hyphen (-).

Allowed characters for Linux user names are (POSIX ("Portable Operating System Interface for Unix") standard (IEEE Standard 1003.1 2008)):
> alphanumeric, including the following common characters: period (.), underscore (_), and hyphen (-).

Therefore, characters that are allowed in IAM user names but not in Linux user names:
> plus (+), equal (=), comma (,), at (@).

This solution will use the following mapping for those special characters when creating users:
* `+` => `.plus.`
* `=` => `.equal.`
* `,` => `.comma.`
* `@` => `.at.`

So instead of `name@email.com` you will need to use `name.at.email.com` when login via SSH.

Linux user names may only be up to 32 characters long.

**Comment by michaelwittig** [comment]
@apatrida any news on this?

**Comment by michaelwittig** [comment]
no feedback received within 3 weeks
